### PR TITLE
Create Mobx-Integration.md at guides

### PR DESCRIPTION
**Pull request by rodrigopivi**
Added some documentation on how is the mobx setup based on the redux guide

### CODE REVIEW

File: docs/guides/Mobx-Integration.md [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+    return (
+      <Provider {...stores} >
+        <AppWithNavigationState />
+      </Provider>
+    );
+  }
+}
+```
+
+And you an also dispatch a navigation action from any component, e.g.:
```
Comment (matthamil) [inline docs/guides/Mobx-Integration.md]: nit: `And you an also` spelling/grammar error

File: docs/guides/Mobx-Integration.md [hunk 2]
```diff
@@ -0,0 +1,76 @@
+# Mobx Integration
+
+Handling your app's navigation state in mobx, is similar to redux. You can pass your own navigation prop to a navigator. Your navigation prop must provide the current state, as well as access to a dispatcher to handle navigation options.
```
Comment (matthamil) [inline docs/guides/Mobx-Integration.md]: nit: mobx is inconsistently capitalized/not capitalized throughout this PR

File: docs/guides/Mobx-Integration.md [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+
+With mobx, your app's state is defined by observable attributes. But react-navigation expects a dispatcher function, to trigger the state changes on our nav state observable:
```
Comment (matthamil) [inline docs/guides/Mobx-Integration.md]: nit: the Mobx official documentation (https://mobx.js.org/#observable-state) refers to these "observable attributes" as "observable properties"

File: docs/guides/Mobx-Integration.md [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+  };
+
+  @autobind
+  @action
+  dispatchNavigation(action, stackState = true) {
+    const previousNavState = stackState ? this.navigationState : null;
+    return this.navigationState = AppNavigator.router.getStateForAction(action, previousNavState);
+  }
+}
+const globalStore = new GlobalStore();
```
Comment (matthamil) [inline docs/guides/Mobx-Integration.md]: nit: new line above `const glovalStore = new GlobalStore();`

File: docs/guides/Mobx-Integration.md [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+
+
```
Comment (matthamil) [inline docs/guides/Mobx-Integration.md]: nit: remove extra newline

File: docs/guides/Mobx-Integration.md [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+```
+<Button
+  onPress={() => this.props.globalStore.dispatchNavigation(
+    NavigationActions.navigate({ routeName: "RouteOne" }),
+  )}
+  title="To Route One"
+/>
+```
+
+Once you do this, your navigation state is stored within your mobx store, at which point you can fire navigation actions using the dispatchNavigation function.
```
Comment (matthamil) [inline docs/guides/Mobx-Integration.md]: nit: Can you wrap dispatchNavigation in back-ticks like so: `dispatchNavigation`?

**Comment by ericvicenti** [comment]
Thank you, @matthamil, for reviewing this doc!

@rodrigopivi, can you review the suggestions? I'm excited to have this doc landed!

Also, be sure to add this page to App.js, so it shows up in the website

File: docs/guides/Mobx-Integration.md [hunk 7]
```diff
@@ -0,0 +1,76 @@
...
+
+Applying this to the example above, you could instead define `AppNavigator` to contain a nested `TabNavigator` as follows:
+
+```js
+const AppNavigator = StackNavigator({
+  Home: { screen: MyTabNavigator },
+});
+```
+
+In this case, once you `inject` and `observe` `AppNavigator` using Mobx as is done in `AppWithNavigationState`, `MyTabNavigator` will automatically have access to navigation state as a `navigation` prop.
```
Comment (hotchpotch) [inline docs/guides/Mobx-Integration.md]: @rodrigopivi 

Your mobx's document is very good!
But I can't work TabNavigator + StackNavigator (nested).

If you can more example or documation, I'm very happy :)

Comment (hotchpotch) [inline docs/guides/Mobx-Integration.md]: I wrote https://github.com/hotchpotch/react-navigation-mobx-example .
Thank you your mobx docs!

**Comment by grabbou** [comment]
Closing as this PR has been left with unaddressed comments for a long time. Please re-open when you have time to work on it. Thank you!

**Pull request closed without merging**